cdap-api/src/main/java/co/cask/cdap/api/app/ApplicationContext.java

```diff
@@ -30,7 +32,8 @@
   /**
    * Get the configuration object.
    *
-   * @return configuration object
+   * @return application configuration provided during application creation, else null
```

Comment:
Changed logic to invoke default Constructor. Hence the user need not check for null.